File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/jms/docs/successhandler/SuccessHandlingProducer.kt
package io.micronaut.jms.docs.successhandler

import io.micronaut.jms.activemq.classic.configuration.ActiveMqClassicConfiguration.CONNECTION_FACTORY_BEAN_NAME
import io.micronaut.jms.annotations.JMSProducer
import io.micronaut.jms.annotations.Queue
import io.micronaut.messaging.annotation.MessageBody

@JMSProducer(CONNECTION_FACTORY_BEAN_NAME)
interface SuccessHandlingProducer {
    @Queue("success-queue")
    fun push(@MessageBody message: String)
}

File: docs-examples/example-python/src/test/python/micronaut/jms/docs/successhandler/SuccessHandlingProducer.py
from abc import ABC, abstractmethod
from typing import Annotated

from micronaut.context.annotation import Requires
from micronaut.jms.annotations import JMSProducer, Queue
from micronaut.messaging.annotation import MessageBody


@Requires(property="spec.name", value="SuccessHandlingSpec")
@JMSProducer("activeMqConnectionFactory")
class SuccessHandlingProducer(ABC):

    @Queue("success-queue")
    @abstractmethod
    def send(self, body: Annotated[str, MessageBody]) -> None:
        ...
